### PR TITLE
[SPARK-42955][SQL] Skip classifyException and wrap AnalysisException for SparkThrowable

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

SparkThrowable is not necessary to be classified by JDBC Dialects and should not be wrapped with a redundantAnalysisException.

 
### Why are the changes needed?

improve error handling for JDBC catalogs

### How was this patch tested?

new unit test.

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @HyukjinKwon

File: sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java [hunk 1]
```diff
@@ -628,6 +630,16 @@ public int hashCode() {
       result = 31 * result + Arrays.hashCode(fieldNames);
       return result;
     }
+
+    @Override
+    public String toString() {
```
Comment (cloud-fan) [inline sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java:635]: can we do it in a separated PR? all `ColumnChange` implementations need this change.

Comment (yaooqinn) [reply sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java:635]: https://github.com/apache/spark/pull/40583/files#diff-8d2ca0d05dd54468ef4faf6ae318ed2288c46be701f3858eecbb371c8f0d420bR343 OK, I will break the test part to line-to-line comparison.

**Review by cloud-fan — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**